src/Features/Core/Portable/Completion/Providers/AbstractAggregateEmbeddedLanguageCompletionProvider.cs

```diff
@@ -107,17 +107,17 @@ public override async Task ProvideCompletionsAsync(CompletionContext context)
         }
 
         public override Task<CompletionChange> GetChangeAsync(Document document, CompletionItem item, char? commitKey, CancellationToken cancellationToken)
-            => GetLanguage(item).CompletionProvider.GetChangeAsync(document, item, commitKey, cancellationToken);
+            => GetLanguage(item).CompletionProvider!.GetChangeAsync(document, item, commitKey, cancellationToken);
```

Comment:
Why the `!`?

Reply:
so, CompletionProvider became nullable.  However, an invariant of this system is that if you have an embedded completion item you can always get back to its original alngauge, which of course means it has a completion provider.

Don't know any way to encode this into the type system though.